hello-security/signup-controller.php

```diff
@@ -0,0 +1,31 @@
+<?php
+include "db-connection.php";
+
+if (strcmp($_POST["role"], "teacher") == 0 || 
+	strcmp($_POST["role"], "student") == 0) {
+
+	$conn = openCon();
+	
+	//prepared statement - defence against SQL injection
+	$stmt = $conn->prepare("INSERT INTO person (email, firstname, password, role) 
+			VALUES (:email, :firstname, :password, :role)");
+
+	$email = htmlentities($_POST["email"]);
+	$firstname = htmlentities($_POST["firstname"]);
+	$passHash = password_hash(htmlentities($_POST["password"]), PASSWORD_DEFAULT);
+	$role = htmlentities($_POST["role"]);
```

Comment:
I like that you extract these in separate variables. It improves readability and changeability.